### DISCUSSION TITLE
Hyprwall, a new GUI wallpaper manager / setter

### DISCUSSION THREAD

**Discussion by ghost** [hyprwm/Hyprland#8157] [Show and tell]
https://github.com/nnyyxxxx/hyprwall

**Hyprwall is a new GUI wallpaper setter for various back ends such as: hyprpaper, swww, swaybg, wallutils, and feh.**

## Differences between other GUI wallpaper pickers:
- **Rust** - Built with Rust, so it's blazingly fast and memory-safe.
- **No dependencies** - Unlike other GUI wallpaper pickers, Hyprwall doesn't have any package dependencies (other than rust), so it's lightweight and easy to install.
- **Minimalist** - Hyprwall is minimalist, the source code is very small compared to other wallpaper pickers e.g. (waypaper).
- **Wrapping** - Hyprwall supports wrapping, so if you choose to you can have a lot of wallpapers shown in the GUI at once (wraps with window size).
- **Performance** - Hyprwall is designed to be performant, it uses a thread pool to load images in parallel and caches images.
- **High capacity** - Hyprwall can handle a large number of wallpapers (over 1000 at one time!) without any issues.
- **Supports swaybg, swww, wallutils, feh, and hyprpaper** - Hyprwall supports a variety of wallpaper backends, so you can use it with your preferred wallpaper tool.

![image](https://github.com/user-attachments/assets/1c6b2aba-54fa-4b98-827c-7caf32c6e818)

**Comment by leoVici9** [comment]
the dependency in the aur packages on rustup iso rust , makes installing far more complex then it should

**Comment by psi4j** [comment]
What happened? It seems this project has moved over to:
https://github.com/MarkusVolk/hyprwall

This was also a part of the old hyprutils? What is this?

**Comment by ghost** [comment]
it was not moved anywhere, that is a fork. I have since deleted the org because I was bored with my life.

**Comment by ghost** [comment]
just use waypaper if you're interested in a gui wallpaper setter

**Reply by psi4j** [reply]
Okay, thanks!
